BGL/doc/BGL/Concepts/HalfedgeGraph.h

```diff
@@ -23,43 +23,68 @@ A model of `HalfedgeGraph` must have the interior property `vertex_point` attach
 
 \cgalHasModel `CGAL::Polyhedron_3`
 \cgalHasModel `CGAL::Surface_mesh`
-
-\cgalHeading{Notations}
-
-<dl>
-<dt>`G`</dt> 	  <dd>A type that is a model of `HalfedgeGraph`.</dd>
-<dt>`g`</dt> 	  <dd>An object of type `G`.</dd>
-<dt>`u`, `v`</dt> <dd>Vertex descriptors.</dd>
-<dt>`e`</dt> 	  <dd>An edge descriptor.</dd>
-<dt>`h`</dt> 	  <dd>A halfedge descriptor.</dd>
-</dl>
-
-\cgalHeading{Associated Types}
-
-Type                                                      | Description
---------------------------------------------------------- | ------------
-`boost::graph_traits<G>::%halfedge_descriptor`             | A `halfedge_descriptor` corresponds to a halfedge in a graph. Must be `DefaultConstructible`, `Assignable`, `EqualityComparable` and `LessThanComparable`.
-
-
-\cgalHeading{Valid Expressions}
-
-Expression                              | Returns                                                                      | Description  
---------------------------------------- | ---------------------------------------------------------------------------- | -----------
-`edge(h, g)`                            | `edge_descriptor`                                                            | The edge corresponding to `h` and `opposite(h)`.
-`halfedge(e, g)`                        | `halfedge_descriptor`                                                        | One of the halfedges corresponding to `e`.
-`halfedge(v, g)`                        | `halfedge_descriptor`                                                        | A halfedge with target `v`. 
-`halfedge(u, v, g)`                     | `std::pair<halfedge_descriptor,bool>`                                        | The halfedge with source `u` and target `v`. The Boolean is `true`, iff this halfedge exists.
-`opposite(h, g)`                        | `halfedge_descriptor`                                                        | The halfedge with source and target swapped.
-`source(h,g)`                           | `vertex_descriptor`                                                          | The source vertex of `h`.
-`target(h,g)`                           | `vertex_descriptor`                                                          | The target vertex of `h`.
-`next(h, g)`                            | `halfedge_descriptor`                                                        | The next halfedge around its face.
-`prev(h, g)`                            | `halfedge_descriptor`                                                        | The previous halfedge around its face.
-`boost::graph_traits<G>::%null_halfedge()` | `halfedge_descriptor`                                                     | Returns a special halfedge that is not equal to any other halfedge.
-
-\cgalHeading{Invariants}
-
-`halfedge(edge(h,g))==h`
```

Comment:
@sloriot, https://github.com/CGAL/cgal/pull/2086#discussion_r116725808: 

    \cgalHeading{Invariants}

    `halfedge(edge(h,g))==h`
> You removed this invariant that is very important.

@afabri: Have you re-added the invariant?